### ISSUE TITLE
Plugin assets not reachable on (lowercase) url's that is generated by htmlHelper

### ISSUE THREAD

**Issue by blieb** [cakephp/cakephp#11379]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.5.4
* Platform and Target: Apache with php 7.0.8 on macOS high sierra. 

### What you did
I've a plugin named "Plugin" and I loaded a css file from plugin in a view (which is inside the Plugin)
$this->Html->css(['Plugin.helpers/upload/uploadInfo'], ['block' => true])

### What happened
it added a css file in html to: 
/plugin/css/helpers/upload/uploadInfo.css

but this url is a 404. 

### What you expected to happen
There are 2 options,
/Plugin/css/helpers/upload/uploadInfo.css << this url works (uppercase P)

I think the best solution is to let the asset middleware also make it possible to use the one with the lowercase plugin name.
When I create symlinks by running "bin/cake plugin assets symlink" it also creates a symlink without the uppercase P.

**Comment by dereuromark** [comment]
It might not just be about upper or lowercase, MyPlugin multiwords would maybe also not be inflected correctly with dash (my-plugin) or underscore (my_plugin), depending on the routing settings.

**Comment by markstory** [comment]
Is your plugin's name actually `Plugin`?

**Comment by blieb** [comment]
Yes don't know why, it is just a small plugin of things used over multiple projects.

**Comment by markstory** [comment]
Ok. Do you have the same issue with plugins not called 'Plugin'?

**Comment by blieb** [comment]
I also have a plugin called "Mobile" but no problems with that one.

**Comment by markstory** [comment]
I wasn't able to reproduce this. 

1. I created a plugin called `Plugin` in `APPROOT/plugins/Plugin` 
2. I added a CSS file `APPROOT/plugins/Plugin/webroot/css/red.css` Which contained `* { color: red !important; }`
3. I loaded the helper in my application's bootstrap.php file using `Plugin::load('Plugin');`
4. In my layout I linked the new css file using `<?= $this->Html->css('Plugin.red') ?>`

The path was right and everything on the page was red as it should be. What did I miss?
